service/server.py

```diff
@@ -161,13 +161,16 @@ def display_title(title_ref):
 def find_titles():
     # TODO: make this method use a WTF form, just like signin()
     if request.method == "POST":
-        search_term = request.form['search_term']
+        search_term = request.form['search_term'].strip()
         LOGGER.info(
             "SEARCH REGISTER: {0} was searched by {1}".format(
                 search_term,
                 current_user.get_id()))
         # Determine search term type and preform search
         title_number_regex = re.compile(TITLE_NUMBER_REGEX)
+        postcode_regex = re.compile(
```

Comment:
The regex could be a global variable. As Python is probably going to compile it once anyway, not really important.